Add documentation to GC Sweep Tests

tyler-cai-microsoft opened the pull request:
[AB#1586](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/1586)

Adds documentation and comments for the GC Sweep Tests

Please give any feedback on anything that may not be well documented enough.

tyler-cai-microsoft committed:
Add documentation

tylerbutler committed:
Change logger signature. (#11636)

The logger interface is used to enable existing build-tools to use the same logging infra as the new tools with minimal changes. After using the interface for a few weeks, I think these changes to the method names remove a lot of duplicate words in code. For example, logger.logVerbose() becomes logger.verbose(), or even log.verbose() if desired.

tylerbutler committed:
[bump] build-tools: 0.3.2000 => 0.4.1000 (major) (#11634)

Post-release major bump of build-tools.

CraigMacomber committed:
Tree: cleanup and use checkout (#11650)

tylerbutler committed:
Fix isLatest calculation in CI and move tool to new CLI (#11628)

CraigMacomber committed:
Tree: fix bench:profile (#11642)

Co-authored-by: Craig Macomber <Craig.Macomber@microsoft.com>

jenn-le committed:
Publish packages in topological order (#11434)

tylerbutler committed:
Start publishing @fluid-tools/build-cli (#11669)

tylerbutler committed:
[bump] build-tools: 0.4.1000 => 0.4.2000 (minor) (#11666)

Post-release minor bump of build-tools.

tylerbutler committed:
[bump] build-tools: 0.4.2000 => 0.4.3000 (minor) (#11674)

CraigMacomber committed:
Tree: Cleanup and document dependencies within tree (#11656)

tylerbutler committed:
Add "none" as a valid option to the generate buildVersion command (#11683)

agarwal-navin commented on packages/test/test-gc-sweep-tests/README.md:
```diff
@@ -2,8 +2,14 @@
-The goal of these tests is to enable testing of the GC sweep infrastructure. Ideally, these tests should run for a few minutes. This test is trying to imitate what might actually happen in the real world and provide enough randomness that it will likely catch any sort of regression or bug missed by more targetted end-to-end tests.
+The goal of these tests is to enable testing of the GC sweep infrastructure. Ideally, these tests should run for a few
+minutes. This test is trying to imitate what might actually happen in the real world and provide enough randomness that
+it will likely catch any sort of regression or bug missed by more targeted end-to-end tests.
```
```suggestion
it will likely catch issues before they reach out to customers.
```

tyler-cai-microsoft committed:
Update fluid-fetch documentation (#11657)

* Update fluid-fetch documentation

agarwal-navin commented on packages/test/test-gc-sweep-tests/src/testDataObjects.ts:
```diff
@@ -30,7 +30,7 @@ import {
 } from "./channelHandler";
 
-// data store that exposes container runtime for testing.
+// The BaseTestDataObject class that exposes useful objects for testing
```
This comment does not convey anything meaningful. It can probably just say that this is the base data object that has basic functionality from which other data objects should derive.

tyler-cai-microsoft committed:
Update packages/test/test-gc-sweep-tests/README.md

Co-authored-by: Navin Agarwal <45832642+agarwal-navin@users.noreply.github.com>

tyler-cai-microsoft committed:
Update comment on GC Sweep Test

agarwal-navin reviewed (approved):
Looks good. You can check this in for now and we can revisit as the test evolves.

tylerbutler committed:
Clean up version-tools APIs and enable api-extractor (#11691)

ghoshkaj committed:
Ensure sockets are not accessed after 4s of inactivity in tinylicious (#11652)

The tinylicious tests infrequently but consistently fail with 502 error code. The node (Axios) server socket timeout is 15s, whereas the underlying socket closes by default after 5s of inactivity. This PR switches to using agentkeepalive package to force the Node server to understand and close connections idle sockets after 4 seconds so that Node doesn't try to use an already closed socket.

jenn-le committed:
Fix object forest (#11682)

The solutions to this type checking issue come with their own issues so it was decided to keep things in their current state and work towards keeping JsonableTree out of public APIs as well as rely on testing to catch these issues.

jzaffiro committed:
Print memory test errors (#11693)

* Print errors at the end of the memory tests

* Consolidate log statements

nedalhy committed:
Improve package building and packing for @fluid-experiemental/property-inspector-table (#11569)

alexvy86 committed:
Processing of execution-time performance test results (#11622)

* New npm scripts to run perf-tests
* New tool to process perf test output files
* Pipeline to run perf benchmarks and send metrics to Kusto
* Update dependencies on @fluid-tools/benchmark to 0.42.0

NicholasCouri committed:
Attempt to address AB 357 (#11645)

* Attempt to address AB 357

alexvy86 committed:
Improvements to benchmark tool (#11661)

- benchmark() and benchmarkMemory() now automatically tag tests with @ExecutionTime or @MemoryUsage, respectively, to allow for selective execution.
- Ability to add a user-specified category to tests, for selective execution.

Specifically for memory tests:
- Ability to define a percentage of samples to use when computing statistics. If not set to 1 (100%), the lowest and highest samples are dropped, and statistics are only computed from the remaining ones.
- Print the computed statistics to the output file when using the custom mocha reporter.
- Unit tests for function that computes statistics.

tylerbutler committed:
Fix and add tests for the buildVersion CLI (#11698)

This change adds tests for the generate buildVersion CLI command and updates the code so that it reads environment variables correctly.

tyler-cai-microsoft committed:
Document GC Sweep Test Object Rundown

tyler-cai-microsoft committed:
Fix gc dataStore no duplicate routes test (#11687)

* Fix gc dataStore no duplicate routes test

tylerbutler committed:
Fix logic with patch flag in `generate buildVersion` (#11701)
